src/main/java/org/openhab/binding/zigbee/converter/ZigBeeConverterColorColor.java

```diff
@@ -106,72 +266,186 @@ public void run() {
                     return;
                 }
 
-                int level;
                 try {
                     if (command instanceof HSBType) {
-                        HSBType hsb = (HSBType) command;
-                        clusterColorControl
-                                .moveToHueAndSaturationCommand((int) (hsb.getHue().doubleValue() * 254.0 / 360.0 + 0.5),
-                                        (int) (hsb.getSaturation().doubleValue() * 254.0 / 100.0 + 0.5), 10)
-                                .get();
-                        level = hsb.getBrightness().intValue();
+                        HSBType color = (HSBType) command;
+                        PercentType brightness = color.getBrightness();
```

Comment:
These two lines are only used within the next if statement so they should be moved into the conditional block. You also have created two variables that aren't used - maybe you can just use the following below -:
```
changeBrightness(((HSBType)command).getBrightness());
```

Reply:
'brightness' is used both in the if statement and in the conditional block. 'color' is used in the next if statement (even if only a typecasting, it is used twice, plus the assignment in 'brightness').

Wouldn't it make sense to keep them then ?